components/template-mgt/org.wso2.carbon.identity.template.mgt.endpoint/src/main/java/org/wso2/carbon/identity/template/mgt/endpoint/impl/TemplatesApiServiceImpl.java

```diff
@@ -47,98 +47,39 @@
 import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.ErrorMessages.ERROR_CODE_UNEXPECTED;
 import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.ErrorMessages.ERROR_CODE_USER_NOT_AUTHORIZED;
 
-public class
-TemplatesApiServiceImpl extends TemplatesApiService {
+@Deprecated
+public class TemplatesApiServiceImpl extends TemplatesApiService {
 
     private static final Log LOG = LogFactory.getLog(TemplatesApiServiceImpl.class);
 
     @Override
     public Response addTemplate(TemplateDTO template) {
 
-        try {
-            TemplateResponseDTO response = postTemplate(template);
```

Comment:
shall we keep the old code as it is without removing? we can return forbidden from the beginning of the function

Reply:
can't keep the old code as it shows error "unreachable statement" when returning forbidden from the beginning of the function.